wayland.c

```diff
@@ -143,6 +175,11 @@ static void seat_handle_capabilities(void *data, struct wl_seat *wl_seat,
 		wl_pointer_add_listener(seat->pointer.wl_pointer,
 			&pointer_listener, seat);
 	}
+	if (capabilities & WL_SEAT_CAPABILITY_TOUCH) {
+		seat->touch.wl_touch = wl_seat_get_touch(wl_seat);
```

Comment:
In case `seat->touch.wl_touch` is already set, we need to release it. See the first `if` block.